### ISSUE TITLE
Please consider make this package Dart 3 compatible

### ISSUE THREAD

**Issue by MagnusJohansson** [kentcb/azure_application_insights#37]
Currently, it depends on http 0.13.3.
It should be upgraded to use the latest Dart 3 compatible [http package 1.0.0](https://pub.dev/packages/http)

```
Because azure_application_insights 3.1.0 depends on http ^0.13.0 and no versions of azure_application_insights match >3.1.0 <4.0.0, azure_application_insights ^3.1.0 requires http ^0.13.0.
So, because my_app depends on both azure_application_insights ^3.1.0 and http ^1.0.0, version solving failed.
```

Thanks.

**Comment by MagnusJohansson** [comment]
Made a PR: #38 , @kentcb  please check

**Comment by MagnusJohansson** [comment]
Regarding the unit testing and mocking, can you consider changing the test to not mock the ByteStream , but rather the StreamedResponse ?
As mentioned in this comment: https://github.com/dart-lang/http/issues/956#issuecomment-1632169699

**Comment by kentcb** [comment]
Resolved by #40 and published
